stepactions/gather-cluster-resources/0.1/gather-cluster-resources.yaml

```diff
@@ -0,0 +1,50 @@
+apiVersion: tekton.dev/v1alpha1
+kind: StepAction
+metadata:
+  name: gather-cluster-resources
+spec:
+  description: >-
+    This StepAction executes the gather extra script that gathers konflux related resources from the cluster.
+    It also allows for a URL with another gather job to be passed to gather custom resources.
+  image: quay.io/konflux-qe-incubator/konflux-qe-tools:latest
+  params:
+    - name: credentials
+      type: string
+      description: A volume containing credentials to the remote cluster
+    - name: kubeconfig
+      type: string
+      description: Relative path to the kubeconfig in the mounted cluster credentials volume
+    - name: ocp-login-command
+      type: string
+      description: Command to log in to the OpenShift cluster
+    - name: gather-url
+      type: list
+      description: URL for the custom resource gathering script. Should be a list of strings
+      default: ()
```

Comment:
is this a valid parameter type? I though we should use `array` with default value `[]`

Reply:
when working with bash, a list is defined as ("url" "url).
Tbh I don't know exactly how it interacts with yaml here and potentially might cause an issue?

Reply:
I think the default value should be `[]`, otherwise the StepAction won't pass the Tekton validation.

See how it's handled e.g. in [buildah Task](https://github.com/konflux-ci/build-definitions/blob/f8da2c780794812e48f56619678e9df958ef0026/task/buildah-remote/0.2/buildah-remote.yaml#L83-L86)

Then the array would be passed to the arguments and handled like:

```yaml
spec:
  params:
    - name: gather-urls
      description: URL for the custom resource gathering script. Should be a list of strings
      type: array
      default: []
  image: quay.io/konflux-qe-incubator/konflux-qe-tools:latest
  args: ["$(params.gather-urls[*])"]
  script: |
    #!/bin/bash
    URLS=( "$@" )
    for url in "${URLS[@]}"
    do
        curl -sSL "${url}" | bash
    done
```

Reply:
valid point indeed! I've made the changes to it but I'm wondering how it should be passed to the step.
I guess if a user would pass the gather_urls it would be something as in yaml right?

["url1", "url2"]
or 
- url1
- url2

right?
just to update the readme

Reply:
yes I think both styles

```yaml
- name: gather-urls
  value: ["url1", "url2"]
```

```yaml
- name: gather-urls
  value:
    - "url1"
    - "url2"
```

should work